core/src/test/java/nl/nn/adapterframework/util/MiscTest.java

```diff
@@ -1,43 +1,610 @@
+        Reader r = new StringReader("<root> \n" +
+                "    <name>GeeksforGeeks</name> \n" +
+                "    <address> \n" +
+                "        <sector>142</sector> \n" +
+                "        <location>Noida</location> \n" +
+                "    </address> \n" +
+                "</root> r");
+        String s = Misc.readerToString(r, "23", true);
+        assertEquals(s, "&lt;root&gt; 23    &lt;name&gt;GeeksforGeeks&lt;/name&gt; 23    &lt;address&gt; 23        &lt;sector&gt;142&lt;/sector&gt; 23        &lt;location&gt;Noida&lt;/location&gt; 23    &lt;/address&gt; 23&lt;/root&gt; r");
+    }
+
+
+
+    /**
+     * Method: replace(String source, String from, String to)
+     */
+    @Test
+    public void testReplace() throws Exception {
+        String a = "Kobe";
+        String res = Misc.replace(a, "Ko", "Phoe");
+        assertEquals(res, "Phoebe");
+    }
+
+    /**
+     * Method: concatStrings(String part1, String separator, String part2)
+     */
+    @Test
+    public void testConcatStrings() throws Exception {
+        String a = "LeBron";
+        String b = "James";
+        String seperator = "//";
+        String res = Misc.concatStrings(a, seperator, b);
+        assertEquals(res, "LeBron//James");
+    }
+
+    /**
+     * Method: hide(String string)
+     */
+    @Test
+    public void testHideString() throws Exception {
+        String a = "test";
+        String res = Misc.hide(a);
+        assertEquals(res, "****");
+    }
+
+    /**
+     * Method: hide(String string, int mode)
+     */
+    @Test
+    public void testHideForStringMode() throws Exception {
+        String a = "test";
+        String res = Misc.hide(a, 1);
+        assertEquals(res, "t**t");
+    }
+
+    /**
+     * Method: byteArrayToString(byte[] input, String endOfLineString, boolean xmlEncode)
+     */
+    @Test
+    public void testByteArrayToString() throws Exception {
+        String s = "test";
+        byte[] arr = s.getBytes();
+        String res = Misc.byteArrayToString(arr, "", true);
+        assertEquals(res, s);
+    }
+
+    /**
+     * Method: gzip(String input)
+     */
+    @Test
+    public void testGzipGUnzipInput() throws Exception {
+        String s = "test";
+        byte[] arr = s.getBytes();
+        byte[] zipped = Misc.gzip(arr);
+        assertEquals(Misc.gunzipToString(zipped), "test");
+    }
+
+    @Test
+    public void testGzipString() throws Exception {
+        String s = "you deserved this";
+        byte[] zipped = Misc.gzip(s);
+        assertEquals(Misc.gunzipToString(zipped), "you deserved this");
+
+    }
+
+    /**
+     * Method: compress(String input)
+     */
+    @Test
+    public void testCompressDecompressInput() throws Exception {
+        String s = "test";
+        byte[] compressed = Misc.compress(s);
+        String decompressed = Misc.decompressToString(compressed);
+        assertEquals(decompressed, "test");
+    }
```

Comment:
Maybe also test the compressed variant to validate if it has actually been compressed

Reply:
I didn't see this yesterday, will take a look at it now.

Reply:
I added a case that tests if the string is compressed. However, I did this by testing the first two indexes of the compressed byte array, which was the same for all compressed Strings, so I assumed this is a sign that it is compressed. Am I correct? Is there another way of testing if it is actually compressed? @nielsm5